home_assistant_streamdeck_yaml.py

```diff
@@ -1114,22 +1443,50 @@ def _update_state(
 ) -> None:
     """Update the state dictionary and update the keys."""
     buttons = config.current_page().buttons
+    dials = config.current_page().dials
     if data["type"] == "event":
         event_data = data["event"]
         if event_data["event_type"] == "state_changed":
             event_data = event_data["data"]
             eid = event_data["entity_id"]
             complete_state[eid] = event_data["new_state"]
 
+            # Handle brightness update
+            if (
+                eid == config.brightness_entity_id
+                and config.brightness_entity_id is not None
+            ):
+                brightness = int(
+                    float(complete_state[config.brightness_entity_id]["state"]),
+                )
+                if brightness >= 0 and brightness <= 100:
+                    console.log(f"Setting default brightness from state {brightness=}")
+                    config.brightness = brightness
+                    if config._is_on:
+                        deck.set_brightness(config.brightness)
+                else:
+                    console.log(f"Invalid brightness state {brightness=}")
+
```

Comment:
shouldn't we return here? you can use an early abort condition for brightness out-of-range